### ISSUE TITLE
connlib on macOS 15 on Intel Macs seemingly hangs after adding candidates

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#7546]
Related #6783 

Connlib candidate discovery seems to hang on macOS 15 on Intel Macs just after receiving the first connection intent. The last relevant log lines (`firezone_tunnel=trace,debug`) are immediately after adding candidates.

```
2024-12-17T08:57:25.228Z DEBUG str0m::ice_::agent: Add new pair CandidatePair(1-3 prio=216172782080229375 state=Waiting attempts=0 unanswered=0 remote=0 last=None nom=None) cid=c1ca2e82-b8b7-40d9-b292-ab800323b0ef
2024-12-17T08:57:25.228Z DEBUG str0m::ice_::agent: Add new pair CandidatePair(3-3 prio=162129582391033854 state=Waiting attempts=0 unanswered=0 remote=0 last=None nom=None) cid=c1ca2e82-b8b7-40d9-b292-ab800323b0ef
```

After this point, connlib is still responsive (we see sentry flush logs after this), but it appears the thread responsible for the candidates has hung.

Will try to find an Intel Mac VM to spin up to test with.

**Comment by thomaseizinger** [comment]
Does it respond to DNS queries? What is the CPU utilisation?

Edit: Ah I see the Slack thread now, investigating.

**Comment by thomaseizinger** [comment]
Turned out to be a bug in MacOS 15.0. Customer managed to fix it by updating to MacOS 15.1
